### ISSUE TITLE
Cooperation: Offline QR Code Generator Firefox add-on

### ISSUE THREAD

**Issue by rugk** [markusfisch/BinaryEye#32]
I am the dev [of a Firefox add-on for generating QR codes](https://github.com/rugk/offline-qr-code) and I really like your QR code scanner/generator, because:

1. It is, as my add-on, offline. (Okay, maybe that's still quite common in Android apps. I can assure you, in add-ons/websites, it is _not_.)
2. (As part of 1.) it does not send tracking data and is FLOSS.
3. It has a really nice icon UI and icon, in contrast to these other barcode generator apps.
4. It works with inverted QR codes and can do useful things with the things it has scanned.

## Now what?

So, however, when generating and scanning a QR code, you always need **two parts**: the reader and scanner. On two devices.

I mostly consider this use case: **transmit some (potentially private) data from your desktop to your phone.**

That's why I'd want to ask, whether we could maybe some kind of "cooperation" to suggest that we can link to each other's app/recommend it, as both things have quite common goals, IMHO. (more on that below)

## What to do?

The exact details can be discussed, but I have a system to show a "random tip" to the add-on user in _rare_ circumstances (non-annoying!). There I already have other tips included, but may also add a recommendation for your app.

## Why?

Because, IMHO, the users of our software(s) will only benefit.

* It helps no one, if you have a private QR code generator, but the QR code scanner sends the data everywhere. (and vice versa)
  So recommending a good app/add-on for the "other" part to the user, helps to increase the overall ("end-to-end") security.
* IMHO, we both put effort into making the UI as simple as possible.
* From my perspective: This app solves many problems (inverted QR codes) that OfflineQr users may get in trouble with. (https://github.com/rugk/offline-qr-code/issues/20)
* My add-on can also be used on Android, but I think the performance and quality of a native app cannot be matched.
* Personally, I see your QR code generator app as the best FLOSS app, I've used. (see list at the top of this issue)

That said (potential problems):
* Obviously, the use cases overlap, so we are "competitors", because a) my add-on works on Android and b) your app can be used to generate QR codes, too. However, as you see above, I'd still use (and I personally do use) your app on Android. Also, we're FLOSS, so it's not that we would gain anything, if people did not use the app or the add-on.
* Problem: The platforms of our pieces of software really _do not_ match. Obviously, this is problematic, because we cannot know whether our users use Android or Firefox.
  IMHO, this is, however, a trade-off we can accept, because it's the nature of QR codes that you want to transmit data between _different_ platforms, devices, browsers etc.

Also, I am indeed calling for an unobtrusive way and not for big banners or so.

---

I am deliberately requesting this in an issue tracker publicly here, to discuss this transparently, because this is as FLOSS projects should do it. That's my opinion, at least. :smiley: 

---

Let me know what you think. :smiley:

**Comment by klawdhfzasjhaa** [comment]
+1
I use those two apps to transfer data between devices. :)

**Comment by rugk** [comment]
Yeah, however, due to lack of response, I do not think this will happen. I've found another very good barcode scanner for Android (on F-Droid and other stores), however: https://github.com/dmitriy-ilchenko/QrAndBarcodeScanner/issues/60

**Comment by markusfisch** [comment]
Sorry for the lack of response - I was quite a bit overwhelmed by your request to be honest 😬 Which is my fault, of course. I just didn't know what to make of it.

But it's good to see you found an alternative 👍 The app looks very good and it may even be a better fit because its UI is way more intuitive than mine is.

**Comment by rugk** [comment]
Well, no problem. It would just have been a small integration, nothing more. :blush: 
Anyway, your app has server me well for many years! And I still have it installed, because I cannot decide sometimes…

But yeah, the other app is very good as well… :shrug:
